homeassistant/components/overkiz/entity.py

```diff
@@ -34,6 +34,14 @@ def __init__(
         self._attr_available = self.device.available
         self._attr_unique_id = self.device.device_url
 
+        if (
+            "#" in self.device_url
+            and not self.device_url.endswith("#1")
+            and self.device.label
```

Comment:
In my case, my devices have always a label that I cannot change.

```JSON
{
  "creationTime": 1496693987000,
  "lastUpdateTime": 1496693987000,
  "label": "Atlantic-4",
  "deviceURL": "io://xxxxxxx/4468654#6",
  "shortcut": false,
  "controllableName": "io:OtherElectricalEnergyConsumptionSensor",
  "definition": {
    "commands": [
      {
        "commandName": "advancedRefresh",
        "nparams": 2
      },
      {
        "commandName": "getSensorName",
        "nparams": 0
      },
      {
        "commandName": "putSensorName",
        "nparams": 1
      }
    ],
    "states": [
      {
        "type": "ContinuousState",
        "qualifiedName": "core:ConsumptionTariff0State"
      },
      {
        "type": "ContinuousState",
        "qualifiedName": "core:ConsumptionTariff1State"
      },
      {
        "type": "ContinuousState",
        "qualifiedName": "core:ConsumptionTariff2State"
      },
      {
        "type": "ContinuousState",
        "qualifiedName": "core:ConsumptionTariff3State"
      },
      {
        "type": "ContinuousState",
        "qualifiedName": "core:ConsumptionTariff4State"
      },
      {
        "type": "ContinuousState",
        "qualifiedName": "core:ConsumptionTariff5State"
      },
      {
        "type": "DiscreteState",
        "values": [
          "good",
          "low",
          "normal",
          "verylow"
        ],
        "qualifiedName": "core:DiscreteRSSILevelState"
      },
      {
        "type": "ContinuousState",
        "qualifiedName": "core:ElectricEnergyConsumptionState"
      },
      {
        "type": "DataState",
        "qualifiedName": "core:NameState"
      },
      {
        "type": "ContinuousState",
        "qualifiedName": "core:RSSILevelState"
      },
      {
        "type": "DiscreteState",
        "values": [
          "dead",
          "lowBattery",
          "maintenanceRequired",
          "noDefect"
        ],
        "qualifiedName": "core:SensorDefectState"
      },
      {
        "type": "DiscreteState",
        "values": [
          "available",
          "unavailable"
        ],
        "qualifiedName": "core:StatusState"
      }
    ],
    "dataProperties": [],
    "widgetName": "CumulativeElectricPowerConsumptionSensor",
    "uiProfiles": [
      "ElectricEnergyConsumption"
    ],
    "uiClass": "ElectricitySensor",
    "qualifiedName": "io:OtherElectricalEnergyConsumptionSensor",
    "type": "SENSOR"
  },
  "states": [
    {
      "name": "core:StatusState",
      "type": 3,
      "value": "available"
    },
    {
      "name": "core:DiscreteRSSILevelState",
      "type": 3,
      "value": "good"
    },
    {
      "name": "core:RSSILevelState",
      "type": 2,
      "value": 100
    },
    {
      "name": "core:ElectricEnergyConsumptionState",
      "type": 1,
      "value": 10344759
    },
    {
      "name": "core:ConsumptionTariff0State",
      "type": 1,
      "value": 10344759
    }
  ],
  "attributes": [
    {
      "name": "core:Manufacturer",
      "type": 3,
      "value": "Atlantic Group"
    },
    {
      "name": "core:MeasurementCategory",
      "type": 3,
      "value": "electric/unknown"
    },
    {
      "name": "core:PowerSourceType",
      "type": 3,
      "value": "mainSupply"
    },
    {
      "name": "core:MeasuredValueType",
      "type": 3,
      "value": "core:ElectricalEnergyInWh"
    }
  ],
  "available": true,
  "enabled": true,
  "placeOID": "xxxxx",
  "widget": "CumulativeElectricPowerConsumptionSensor",
  "measurementCategory": "ELECTRIC_UNKNOWN",
  "type": 2,
  "oid": "xxxx",
  "uiClass": "ElectricitySensor"
}
```

Reply:
In my case I could not rename the main device (HP_Actuator for me, Atlantic for you).
Instead I have an "Atlantic/Thermor/Sauter tab" (after the Sonos on your cas) where I find all my sub devices and where I can rename them.
The only thing I can think of:  
Detect if the provided label is "MAIN_DEVICE-XXX" or not.  
If not, use the provided label.
if it is, use the entity_description like we have

Reply:
Well actually, you could rename a device or sub device with API, even if there is no UI for that.  

I just renamed my main device with a call like:  
`PUT https://www.tahomalink.com/enduser-mobile-web/enduserAPI/setup/devices/URL_ENCODED_DEVICE_URL/URL_ENCODED_NEW_NAME`
with all the credentials of course.

But it's not what user could do, so we have to figure out a solution

Reply:
@nyroDev in the past we always used the name of the parent device, for all sub devices. If some devices don't have a nice parent device name, I don't think we should add workaround / hacks for that. Agree? :)

Reply:
@tetienne I just pushed something that should works for you too.

@iMicknl it's not true, on my prod installation, I do have sub device with great name. 
On my case on current prod version, the name are "SUB_DEVICE_NAME SENSOR" where :
- SUB_DEVICE_NAME is the name coming from Overkiz API
- SENSOR is coming from the code to add the kinf od the sensor (Temperature for instance)

With the change you did in order to respect the new naming convention, on my case, I have 3 Temperature sensors wth the exact same name.

With the last push I did, I try to respect all of that:
- If the user didn't (or couldn't) rename sub entites (like @tetienne), the name will be "MASTER_NAME-XXX". in that case, nothing is done and the SENSOR name will be sued
- If the user did rename sub entites (like me), the name he user gave is used, without adding the SENSOR at the end.

If I understood correctly the new naming convention, this is correct.

Reply:
I still wonder why my fix broke your behavior. I am not sure if we ever supported your behavior.

Can you share your diagnostics file here? (preferably without any changes made through endpoints that are not accessible for users via the Somfy app).

@nyroDev can you perhaps mark this as a draft for now, until we all agree on the next steps :).

Reply:
Here is a screenshot of my prod installation, no manual renaming was done in HA    
![Capture d’écran de 2022-09-29 10-05-21](https://user-images.githubusercontent.com/265662/192975997-53a6d37d-999d-4118-a269-4e5a167de9a6.png)

On dev, the last 3 lines are "Temperature"

Reply:
With current dev :  
![Capture d’écran de 2022-09-29 10-13-09](https://user-images.githubusercontent.com/265662/192977670-40fc2e64-ea36-4c11-a030-2d6810f826aa.png)

The log of the dev:  
[config_entry-overkiz-0807d4e538672f5e8cf71f4a1e7080eb.json.txt](https://github.com/home-assistant/core/files/9672535/config_entry-overkiz-0807d4e538672f5e8cf71f4a1e7080eb.json.txt)

Here is an extract of setup.json from overkiz to show labels (obfuscation is mine) :  

```
{
    "creationTime": 1592393180000,
    "lastUpdateTime": 1592393180000,
    "label": "Pompe \u00e0 chaleur",
    "deviceURL": "io:\/\/XXXXXX\/2674179#1",
    "shortcut": false,
    "controllableName": "io:AtlanticPassAPCHeatPumpMainComponent",
    "definition": {
        "commands": [{
            "commandName": "advancedRefresh",
            "nparams": 2
        }, {
            "commandName": "cancelAbsence",
            "nparams": 0
        }, {
            "commandName": "delayedStopIdentify",
            "nparams": 1
        }, {
            "commandName": "getName",
            "nparams": 0
        }, {
            "commandName": "identify",
            "nparams": 0
        }, {
            "commandName": "refreshDeviceSerialNumber",
            "nparams": 0
        }, {
            "commandName": "refreshElectricEnergyConsumption",
            "nparams": 0
        }, {
            "commandName": "refreshErrorCode",
            "nparams": 0
        }, {
            "commandName": "refreshHeatingDerogationAvailability",
            "nparams": 0
        }, {
            "commandName": "refreshOperatingMode",
            "nparams": 0
        }, {
            "commandName": "refreshOutsideTemperature",
            "nparams": 0
        }, {
            "commandName": "refreshTimeProgramById",
            "nparams": 1
        }, {
            "commandName": "refreshZonesNumber",
            "nparams": 0
        }, {
            "commandName": "refreshZonesTargetTemperature",
            "nparams": 0
        }, {
            "commandName": "setAbsenceCoolingTargetTemperature",
            "nparams": 1
        }, {
            "commandName": "setAbsenceEndDateTime",
            "nparams": 1
        }, {
            "commandName": "setAbsenceHeatingTargetTemperature",
            "nparams": 1
        }, {
            "commandName": "setAbsenceStartDateTime",
            "nparams": 1
        }, {
            "commandName": "setName",
            "nparams": 1
        }, {
            "commandName": "setTimeProgramById",
            "nparams": 2
        }, {
            "commandName": "startIdentify",
            "nparams": 0
        }, {
            "commandName": "stopIdentify",
            "nparams": 0
        }, {
            "commandName": "wink",
            "nparams": 1
        }, {
            "commandName": "pairOneWayController",
            "nparams": 2
        }, {
            "commandName": "refreshAbsenceSchedulingAvailability",
            "nparams": 0
        }, {
            "commandName": "refreshDHWAvailability",
            "nparams": 0
        }, {
            "commandName": "refreshEnergyConsumptionAvailability",
            "nparams": 0
        }, {
            "commandName": "refreshOutsideTemperatureSensorAvailability",
            "nparams": 0
        }, {
            "commandName": "refreshProductType",
            "nparams": 0
        }, {
            "commandName": "refreshThermalSchedulingAvailability",
            "nparams": 0
        }, {
            "commandName": "refreshZonesPassAPCCoolingProfile",
            "nparams": 0
        }, {
            "commandName": "refreshZonesPassAPCHeatingProfile",
            "nparams": 0
        }, {
            "commandName": "refreshZonesTemperature",
            "nparams": 0
        }, {
            "commandName": "refreshZonesTemperatureSensorAvailability",
            "nparams": 0
        }, {
            "commandName": "refreshZonesThermalConfiguration",
            "nparams": 0
        }, {
            "commandName": "setPassAPCOperatingMode",
            "nparams": 1
        }, {
            "commandName": "unpairAllOneWayControllers",
            "nparams": 0
        }, {
            "commandName": "unpairOneWayController",
            "nparams": 2
        }],
        "states": [{
            "type": "ContinuousState",
            "qualifiedName": "core:AbsenceCoolingTargetTemperatureState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:AbsenceEndDateTimeState"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:AbsenceHeatingTargetTemperatureState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:AbsenceStartDateTimeState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:DeviceSerialNumberState"
        }, {
            "type": "DiscreteState",
            "values": ["good", "low", "normal", "verylow"],
            "qualifiedName": "core:DiscreteRSSILevelState"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:ECSPowerConsumptionExtra1State"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:ECSPowerConsumptionExtra2State"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:ECSPowerConsumptionExtra3State"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:ECSPowerConsumptionState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:ErrorCodeState"
        }, {
            "type": "DiscreteState",
            "values": ["available", "unavailable"],
            "qualifiedName": "core:HeatingDerogationAvailabilityState"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:HeatingPowerConsumptionExtra1State"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:HeatingPowerConsumptionExtra2State"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:HeatingPowerConsumptionExtra3State"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:HeatingPowerConsumptionState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:NameState"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:PriorityLockTimerState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:ProductModelNameState"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:RSSILevelState"
        }, {
            "type": "DiscreteState",
            "values": ["available", "unavailable"],
            "qualifiedName": "core:StatusState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:TimeProgram1State"
        }, {
            "type": "DataState",
            "qualifiedName": "core:TimeProgram2State"
        }, {
            "type": "DataState",
            "qualifiedName": "core:TimeProgram3State"
        }, {
            "type": "DataState",
            "qualifiedName": "core:TimeProgram4State"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:VersionState"
        }, {
            "type": "DataState",
            "qualifiedName": "core:ZonesNumberState"
        }, {
            "type": "DiscreteState",
            "values": ["available", "unavailable"],
            "qualifiedName": "io:AbsenceSchedulingAvailabilityState"
        }, {
            "type": "DiscreteState",
            "values": ["dateScheduling", "numberOfDaysScheduling"],
            "qualifiedName": "io:AbsenceSchedulingModeState"
        }, {
            "type": "DiscreteState",
            "values": ["available", "unavailable"],
            "qualifiedName": "io:DHWAvailabilityState"
        }, {
            "type": "DiscreteState",
            "values": ["available", "unavailable"],
            "qualifiedName": "io:EnergyConsumptionAvailabilityState"
        }, {
            "type": "DiscreteState",
            "values": ["cooling", "drying", "heating", "stop"],
            "qualifiedName": "io:LastPassAPCOperatingModeState"
        }, {
            "type": "DiscreteState",
            "values": ["cooling", "drying", "heating", "stop"],
            "qualifiedName": "io:PassAPCOperatingModeState"
        }, {
            "type": "DiscreteState",
            "values": ["accumulationDomesticHotWater", "airConditioning", "boiler", "convector", "doubleFlowControlledMechanicalVentilation", "heatPump", "heater", "hybrid", "singleFlowControlledMechanicalVentilation", "thermodynamicDomesticHotWater", "zoneController"],
            "qualifiedName": "io:PassAPCProductTypeState"
        }, {
            "type": "DiscreteState",
            "values": ["comfortLevel1", "comfortLevel2", "comfortLevel3", "comfortLevel4", "environmentProtection", "humanProtection", "userLevel1", "userLevel2"],
            "qualifiedName": "io:PriorityLockLevelState"
        }, {
            "type": "DiscreteState",
            "values": ["LSC", "SAAC", "SFC", "UPS", "externalGateway", "localUser", "myself", "rain", "security", "temperature", "timer", "user", "wind"],
            "qualifiedName": "io:PriorityLockOriginatorState"
        }, {
            "type": "DiscreteState",
            "values": ["available", "unavailable"],
            "qualifiedName": "io:ThermalSchedulingAvailabilityState"
        }, {
            "type": "DiscreteState",
            "values": ["heatingAndCoolingCommonScheduling", "heatingAndCoolingSeparatedScheduling"],
            "qualifiedName": "io:ThermalSchedulingModeState"
        }],
        "dataProperties": [{
            "value": "500",
            "qualifiedName": "core:identifyInterval"
        }],
        "widgetName": "AtlanticPassAPCHeatPump",
        "uiProfiles": ["Specific"],
        "uiClass": "HeatingSystem",
        "uiClassifiers": ["generator"],
        "qualifiedName": "io:AtlanticPassAPCHeatPumpMainComponent",
        "type": "ACTUATOR"
    },
    "states": [{
        "name": "core:NameState",
        "type": 3,
        "value": "HP_Actuator"
    }, {
        "name": "core:VersionState",
        "type": 3,
        "value": "43363637323038202020"
    }, {
        "name": "core:StatusState",
        "type": 3,
        "value": "available"
    }, {
        "name": "core:DiscreteRSSILevelState",
        "type": 3,
        "value": "normal"
    }, {
        "name": "core:RSSILevelState",
        "type": 2,
        "value": 68
    }, {
        "name": "core:ProductModelNameState",
        "type": 3,
        "value": "R002412300"
    }, {
        "name": "io:PassAPCProductTypeState",
        "type": 3,
        "value": "heatPump"
    }, {
        "name": "core:ZonesNumberState",
        "type": 1,
        "value": 2
    }, {
        "name": "io:DHWAvailabilityState",
        "type": 3,
        "value": "available"
    }, {
        "name": "core:HeatingPowerConsumptionState",
        "type": 1,
        "value": 10
    }, {
        "name": "core:HeatingPowerConsumptionExtra1State",
        "type": 1,
        "value": 3
    }, {
        "name": "core:HeatingPowerConsumptionExtra2State",
        "type": 1,
        "value": 3
    }, {
        "name": "core:HeatingPowerConsumptionExtra3State",
        "type": 1,
        "value": 0
    }, {
        "name": "core:ECSPowerConsumptionState",
        "type": 1,
        "value": 10
    }, {
        "name": "core:ECSPowerConsumptionExtra1State",
        "type": 1,
        "value": 1
    }, {
        "name": "core:ECSPowerConsumptionExtra2State",
        "type": 1,
        "value": 0
    }, {
        "name": "core:ECSPowerConsumptionExtra3State",
        "type": 1,
        "value": 0
    }, {
        "name": "io:EnergyConsumptionAvailabilityState",
        "type": 3,
        "value": "available"
    }, {
        "name": "io:PassAPCOperatingModeState",
        "type": 3,
        "value": "heating"
    }, {
        "name": "core:HeatingDerogationAvailabilityState",
        "type": 3,
        "value": "available"
    }, {
        "name": "io:ThermalSchedulingAvailabilityState",
        "type": 3,
        "value": "available"
    }, {
        "name": "io:ThermalSchedulingModeState",
        "type": 3,
        "value": "heatingAndCoolingSeparatedScheduling"
    }, {
        "name": "core:TimeProgram1State",
        "type": 10,
        "value": [{
            "monday": [{
                "start": "06:00",
                "end": "23:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "tuesday": [{
                "start": "06:00",
                "end": "23:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "wednesday": [{
                "start": "06:00",
                "end": "23:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "thursday": [{
                "start": "06:00",
                "end": "23:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "friday": [{
                "start": "06:00",
                "end": "23:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "saturday": [{
                "start": "06:00",
                "end": "23:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "sunday": [{
                "start": "06:00",
                "end": "23:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }]
    }, {
        "name": "core:TimeProgram2State",
        "type": 10,
        "value": [{
            "monday": [{
                "start": "00:00",
                "end": "24:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "tuesday": [{
                "start": "00:00",
                "end": "24:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "wednesday": [{
                "start": "00:00",
                "end": "24:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "thursday": [{
                "start": "00:00",
                "end": "24:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "friday": [{
                "start": "00:00",
                "end": "24:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "saturday": [{
                "start": "00:00",
                "end": "24:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "sunday": [{
                "start": "00:00",
                "end": "24:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }]
    }, {
        "name": "core:TimeProgram3State",
        "type": 10,
        "value": [{
            "monday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "tuesday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "wednesday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "thursday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "friday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "saturday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "sunday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }]
    }, {
        "name": "core:TimeProgram4State",
        "type": 10,
        "value": [{
            "monday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "tuesday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "wednesday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "thursday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "friday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "saturday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }, {
            "sunday": [{
                "start": "06:00",
                "end": "22:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }, {
                "start": "00:00",
                "end": "00:00"
            }]
        }]
    }, {
        "name": "io:AbsenceSchedulingAvailabilityState",
        "type": 3,
        "value": "available"
    }, {
        "name": "io:AbsenceSchedulingModeState",
        "type": 3,
        "value": "dateScheduling"
    }, {
        "name": "core:AbsenceHeatingTargetTemperatureState",
        "type": 2,
        "value": 8
    }, {
        "name": "core:AbsenceCoolingTargetTemperatureState",
        "type": 2,
        "value": 35
    }, {
        "name": "core:ErrorCodeState",
        "type": 1,
        "value": 0
    }, {
        "name": "io:LastPassAPCOperatingModeState",
        "type": 3,
        "value": "heating"
    }],
    "attributes": [{
        "name": "core:Manufacturer",
        "type": 3,
        "value": "Atlantic Group"
    }, {
        "name": "core:FirmwareRevision",
        "type": 3,
        "value": "C667208"
    }, {
        "name": "core:ElectricPowerConsumptionStateMeasuredValueType",
        "type": 3,
        "value": "core:ElectricalPowerInkW"
    }],
    "available": true,
    "enabled": true,
    "placeOID": "XXXXX",
    "widget": "AtlanticPassAPCHeatPump",
    "type": 1,
    "oid": "XXXXX",
    "uiClass": "HeatingSystem"
},
{
    "creationTime": 1592393184000,
    "lastUpdateTime": 1592393184000,
    "label": "Temp RDC",
    "deviceURL": "io:\/\/XXXXXX\/2674179#11",
    "shortcut": false,
    "controllableName": "io:AtlanticPassAPCZoneTemperatureSensor",
    "definition": {
        "commands": [],
        "states": [{
            "type": "DiscreteState",
            "values": ["available", "unavailable"],
            "qualifiedName": "core:StatusState"
        }, {
            "type": "ContinuousState",
            "qualifiedName": "core:TemperatureState"
        }],
        "dataProperties": [],
        "widgetName": "TemperatureSensor",
        "uiProfiles": ["Temperature"],
        "uiClass": "TemperatureSensor",
        "qualifiedName": "io:AtlanticPassAPCZoneTemperatureSensor",
        "type": "SENSOR"
    },
    "states": [{
        "name": "core:StatusState",
        "type": 3,
        "value": "available"
    }, {
        "name": "core:TemperatureState",
        "type": 2,
        "value": 21.9
    }],
    "attributes": [{
        "name": "core:Manufacturer",
        "type": 3,
        "value": "Atlantic Group"
    }, {
        "name": "core:FirmwareRevision",
        "type": 3,
        "value": "C667208"
    }, {
        "name": "core:MeasuredValueType",
        "type": 3,
        "value": "core:TemperatureInCelcius"
    }, {
        "name": "core:PowerSourceType",
        "type": 3,
        "value": "mainSupply"
    }, {
        "name": "core:MaxSensedValue",
        "type": 2,
        "value": 30
    }, {
        "name": "core:MinSensedValue",
        "type": 2,
        "value": 5
    }],
    "available": true,
    "enabled": true,
    "placeOID": "XXXXX",
    "widget": "TemperatureSensor",
    "type": 2,
    "oid": "XXXXX",
    "uiClass": "TemperatureSensor"
}
```

Reply:
The last commit should fix this.